### PR TITLE
Run singularity

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
**Description of the Pull Request (PR):**

Create a `run-singularity` script to enable the functionality 
```
$ ./lolcow
 ________________________________________
/ You will be winged by an anti-aircraft \
\ battery.                               /
 ----------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

```

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [x] I have tested this PR locally with a `make test`
- [x] This PR is against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

File: mlocal/frags/Makefile.stub [hunk 1]
```diff
@@ -79,6 +81,12 @@ $(starter_suid_INSTALL): $(starter)
 	$(V)install -d $(@D)
 	$(V)install -m 4755 $(starter) $(starter_suid_INSTALL)
 
+# Run-Singularity bin script
+$(dist_bin_SCRIPTS_INSTALL): $(dist_bin_SCRIPTS)
+	@echo " INSTALL" $@
```
Comment (jmstover) [inline mlocal/frags/Makefile.stub:86]: *EDIT:* Ignore this ...

Change this to:
`@echo " INSTALL run-singularity"`

**Review by jmstover — DISMISSED** [review]

**Review by jmstover — DISMISSED** [review]
@yhcote should look over

**Comment by ArangoGutierrez** [comment]
@jmstover added a new commit. ;)

File: mlocal/frags/Makefile.stub [hunk 2]
```diff
@@ -34,9 +36,9 @@ abs_BUILDDIR := `readlink -f $(BUILDDIR)`
 cgo_CPPFLAGS = -I$(BUILDDIR) -I$(SOURCEDIR)/src/runtime -I$(SOURCEDIR)/src/runtime/c/lib -include $(abs_BUILDDIR)/config.h
 cgo_LDFLAGS = -L$(abs_BUILDDIR)/lib -L$(BUILDDIR) -lruntime
 
-INSTALLFILES := $(singularity_INSTALL) $(starter_INSTALL) $(starter_suid_INSTALL) $(sessiondir) $(config_INSTALL)
+INSTALLFILES := $(singularity_INSTALL) $(starter_INSTALL) $(starter_suid_INSTALL) $(sessiondir) $(config_INSTALL) $(dist_bin_SCRIPTS_INSTALL)
 
-CLEANFILES += $(libruntime) $(starter) $(singularity) $(go_BIN) $(go_OBJ)
+CLEANFILES += $(libruntime) $(starter) $(singularity) $(go_BIN) $(go_OBJ) $(dist_bin_SCRIPTS_INSTALL)
```
Comment (jmstover) [inline mlocal/frags/Makefile.stub]: You shouldn't need this here... this should be for files under `$(BUILDDIR)` ... which you don't have in your case. What would happen is:  `make clean`  would delete the file from where it was under `$(PREFIX)`

**Review by jmstover — CHANGES_REQUESTED** [review]

File: mlocal/frags/Makefile.stub (hunk 2 above)
Comment (yhcote) [reply mlocal/frags/Makefile.stub]: Lol I am just trying to resolve conflicts for this PR before I can review it. But thanks for joining in. And even without having read the patch yet, I think you're right. Gimme another minute ;)

File: scripts/run-singularity [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
```
Comment (yhcote) [inline scripts/run-singularity:1]: Is this an original file coming from 2.X ?

File: scripts/run-singularity [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# to reproduce, distribute copies to the public, prepare derivative works, and
+# perform publicly and display publicly, and to permit other to do so. 
+# 
+# 
+
+# Simple wrapper for 'singularity run'
+
+exec /usr/bin/env singularity --quiet run "$@"
+
+exit 255
```
Comment (yhcote) [inline scripts/run-singularity]: Not a big deal but the end of the file is not an expected unix text file ending.

File: mlocal/frags/Makefile.stub (hunk 1 above)
Comment (yhcote) [reply mlocal/frags/Makefile.stub:86]: yeah this is correct.

**Review by yhcote — CHANGES_REQUESTED** [review]
Just a few things here and there and what @jmstover already pointed out.

File: scripts/run-singularity (hunk 3 above)
Comment (jmstover) [reply scripts/run-singularity:1]: Yes, this should be exactly what `bin/run-singularity` is in the 2.x branch

Comment (ArangoGutierrez) [reply scripts/run-singularity:1]: +1

**Review by jmstover — APPROVED** [review]

**Review by bauerm97 — APPROVED** [review]

**Review by GodloveD — COMMENTED** [review]
lgtm

Thanks @ArangoGutierrez !

**Review by GodloveD — APPROVED** [review]

**Comment by ArangoGutierrez** [comment]
@yhcote RFR

**Review by yhcote — APPROVED** [review]
Sorry for holding up that one! LGTM

**Pull request merged**